### ISSUE TITLE
reverseGeocode location object not working

### ISSUE THREAD

**Issue by jcreighton** [mapbox/mapbox-sdk-js#40]
First, I think the longitude & latitude are in the wrong order in the constant:
```module.exports.API_GEOCODER_REVERSE = '/v4/geocode/{dataset}/{location.longitude},{location.latitude}.json';```

Second, when sending in an object {latitude: lat, longitude: long}, the query string isn't properly set due to an issue in template.js in rest/interceptor. The uri is set using param[value] but this doesn't work then the value is "location.longitude", etc.

I'm fixing this right now by explicitly setting long & lat params in geocoder.js like so:

 ```   params: {
      latitude: location.latitude,
      longitude: location.longitude,
      dataset: dataset
    }```

and rewriting the constant to: 

```module.exports.API_GEOCODER_REVERSE = '/v4/geocode/{dataset}/{latitude},{longitude}.json';```

**Comment by tmcw** [comment]
The order should be longitude, latitude - [that's what's in the docs](https://www.mapbox.com/developers/api/geocoding/#reverse) and a query like [-73.989,40.733](https://api.mapbox.com/v4/geocode/mapbox.places/-73.989,40.733.json?access_token=pk.eyJ1IjoidG1jdyIsImEiOiJIZmRUQjRBIn0.lRARalfaGHnPdRcc-7QZYQ) works properly, whereas the [reverse](https://api.mapbox.com/v4/geocode/mapbox.places/40.733,-73.989.json?access_token=pk.eyJ1IjoidG1jdyIsImEiOiJIZmRUQjRBIn0.lRARalfaGHnPdRcc-7QZYQ) finds antarctica.

As far as the bad token replacement, I think you nailed it. Tagging as a bug.

**Comment by jcreighton** [comment]
Ah yes, you're right about the long/lat order. Thanks for that!

**Comment by tmcw** [comment]
:+1: thanks for the bug report, this is fixed & published in 0.8.2
